### PR TITLE
Add support for getting and accepting quotes

### LINKED ISSUE #78: Add support for amount-less chain swaps
Please add support for https://github.com/BoltzExchange/boltz-backend/pull/659

### PR DESCRIPTION
**Pull request by ok300**
This PR adds the ability to get and accept quotes for Zero-Amount Chain Swaps.

Fixes #78

### CODE REVIEW

File: src/swaps/boltz.rs [hunk 1]
```diff
@@ -17,10 +17,10 @@
 //!     output_amount - base_fees - claim_fee
 //! );
 
-use bitcoin::key;
 use bitcoin::{
     hashes::sha256, hex::DisplayHex, taproot::TapLeaf, PublicKey, ScriptBuf, Transaction,
 };
+use bitcoin::{key, Amount};
```
Comment (michael1011) [inline src/swaps/boltz.rs]: Why is `Amount` being imported?

File: src/swaps/boltz.rs [hunk 2]
```diff
@@ -564,6 +564,28 @@ impl BoltzApiClientV2 {
         let end_point = format!("chain/{}/transaction", chain);
         Ok(serde_json::from_str(&self.post(&end_point, data)?)?)
     }
+
+    /// Looks up the quote for a Zero-Amount Receive Chain Swap.
```
Comment (michael1011) [inline src/swaps/boltz.rs]:
```suggestion
    /// Gets a quote for a Zero-Amount or over- or underpaid Chain Swap.
```

File: src/swaps/boltz.rs [hunk 3]
```diff
@@ -564,6 +564,28 @@ impl BoltzApiClientV2 {
...
+    ///
+    /// If the user locked up a valid amount, it will return the server lockup amount. In all other
+    /// cases, it will return an error.
+    pub fn get_quote(&self, swap_id: &str) -> Result<GetQuoteResponse, Error> {
+        let end_point = format!("swap/chain/{swap_id}/quote");
+        Ok(serde_json::from_str(&self.get(&end_point)?)?)
+    }
+
+    /// Accepts a specific quote for a Zero-Amount Receive Chain Swap.
```
Comment (michael1011) [inline src/swaps/boltz.rs]:
```suggestion
    /// Accepts a specific quote for a Zero-Amount or over- or underpaid Chain Swap.
```

File: src/swaps/boltz.rs (hunk 1 above)
Comment (ok300) [reply src/swaps/boltz.rs]: You're right, it's from an earlier code version where

```rust
pub fn accept_quote(&self, swap_id: &str, amount_sat: u64)
```

took an `Amount` as arg. I figured that's overkill, so I reverted to `u64`. Removing the extra import.

**Review by michael1011 — APPROVED** [review]

**Comment by ok300** [comment]
Could this be merged?

We'll release zero-amount swaps very soon, so we'll need your mainline `trunk` to have the changes in this PR.

**Pull request merged**